crates/store/src/db/models/queries/accounts.rs

```diff
@@ -494,7 +494,7 @@ impl TryFrom<AccountVaultUpdateRaw> for AccountVaultValue {
     type Error = DatabaseError;
 
     fn try_from(raw: AccountVaultUpdateRaw) -> Result<Self, Self::Error> {
-        let vault_key = Word::read_from_bytes(&raw.vault_key)?;
+        let vault_key = AssetVaultKey::new_unchecked(Word::read_from_bytes(&raw.vault_key)?);
```

Comment:
Not for this PR, but we should probably have a way to serialize/deserialize `AssetVaultKey`. cc @PhilippGackstatter.